src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -276,8 +290,11 @@ private void RefreshPortDefaultValueMarkerVisible()
         /// </summary>
         protected override void RefreshPortColors()
         {
-            //Is in function state
-            if (node.NodeModel.IsPartiallyApplied)
+            //This variable checks if the node is a function class
+            isFunctionNode = node.NodeModel.CachedValue != null &&
+                node.NodeModel.CachedValue.IsFunction;
```

Comment:
Shouldn't this be:
```
isFunctionNode = node.NodeModel.IsPartiallyApplied || (node.NodeModel.CachedValue != null &&
                node.NodeModel.CachedValue.IsFunction);
```

Reply:
`node.NodeModel.IsPartiallyApplied ` will not make sense for now because there are cases where the node doesn't return a function ever, like the python node.

Reply:
hmm, I actually think it should be
```
isFunctionNode = node.NodeModel.IsPartiallyApplied && (node.NodeModel.CachedValue != null &&
                node.NodeModel.CachedValue.IsFunction);```

DS is a dynamic language, we could probably write a function which returns another function, and it would now turn ports blue, even if it was not partially unsatisfied.

Reply:
@filipeotero the check was `if (node.NodeModel.IsPartiallyApplied)` earlier, so by removing this check, won't you break behavior? I think this check is necessary for those nodes for which you can determine their "unsatisfied" state even before execution.

Reply:
I mean a case like this that was working earlier in manual mode will fail won't it:
![default_value](https://user-images.githubusercontent.com/5710686/165845559-62888986-dd21-4cea-90a7-be13253fd5b5.gif)